cmd/peer-rest-server.go

```diff
@@ -701,6 +701,62 @@ func (s *peerRESTServer) PutBucketNotificationHandler(w http.ResponseWriter, r *
 	w.(http.Flusher).Flush()
 }
 
+// Return disk IDs of all the local disks.
+func getLocalDiskIDs(z *xlZones) []string {
+	var ids []string
+
+	for zoneIdx := range z.zones {
+		for _, set := range z.zones[zoneIdx].sets {
+			disks := set.getDisks()
+			for _, disk := range disks {
+				if disk == nil {
+					continue
+				}
+				if disk.Hostname() == "" {
+					id, err := disk.GetDiskID()
+					if err != nil {
+						continue
+					}
+					if id == "" {
+						continue
+					}
+					ids = append(ids, id)
+				}
+			}
+		}
+	}
+
+	return ids
+}
+
+// GetLocalDiskIDs - Return disk IDs of all the local disks.
+func (s *peerRESTServer) GetLocalDiskIDs(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		s.writeErrorResponse(w, errors.New("Invalid request"))
+		return
+	}
+
+	ctx := newContext(r, w, "GetLocalDiskIDs")
+
+	objLayer := newObjectLayerWithoutSafeModeFn()
+
+	// Service not initialized yet
```

Comment:
We actually have error code for this errServiceNotInitialized you can use that..